### PR TITLE
Fixes clicking things under windows/doors.

### LINKED ISSUE #27287: i can greytide grilles from under windows
https://gyazo.com/0a2336fc01b47eeaccf131b8eccbe28c

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #27287

### CODE REVIEW

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -141,6 +144,22 @@
 		else
 			RangedAttack(A,params)
 
+//Is the atom obscured by a PREVENT_CLICK_UNDER object above it
+/atom/proc/IsObscured()
+	if(!isturf(loc)) //This only makes sense for things directly on turfs for now
+		return FALSE
+	var/turf/T = loc
+	for(var/atom/movable/AM in T)
+		if(AM.flags & PREVENT_CLICK_UNDER && AM.layer > layer)
```
Comment (MrPerson) [inline code/_onclick/click.dm]: You should either remove the flag when a door opens or bake a density check into this line and the turf line below so people can click on stuff underneath an open door.

Comment (AnturK) [inline code/_onclick/click.dm]: Good point, the doors were mostly afterthought.

**Comment by kevinz000** [comment]
wtf clicking things under doors was a robustness thing ;-;

**Comment by WJohn** [comment]
this bug will continue to exist in some form or another until the end of time.

**Comment by AnturK** [comment]
This is good now.

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -141,6 +144,22 @@
 		else
 			RangedAttack(A,params)
 
+//Is the atom obscured by a PREVENT_CLICK_UNDER object above it
+/atom/proc/IsObscured()
+	if(!isturf(loc)) //This only makes sense for things directly on turfs for now
+		return FALSE
+	var/turf/T = loc
+	for(var/atom/movable/AM in T)
+		if(AM.flags & PREVENT_CLICK_UNDER && AM.density && AM.layer > layer)
```
Comment (PsiOmegaDelta) [inline code/_onclick/click.dm]: I thought this looked interesting enough to port, but noted that there should probably be a ```plane``` check here as well.

**Pull request merged**